### DISCUSSION TITLE
POST Read Callback never called.

### DISCUSSION THREAD

**Discussion by jshanab** [curl/curl#7356] [Q&A] [answered]
I have been using curl to implement rtsp-over-http/s for years. We have been using 7.48.x and the Post was using CURLOPT_READFUNCTION and CURLOPT_READDATA eith a content length set to a very very large value.

The protocol is such that you open the post channel and it stays open and all rtsp commands are sent base 64 encoded. 
We use the Pause and Resume to handle this and once video is streaming on the GET channel, a command is only sent every 5 seconds until the TEARDOWN is sent.

We are updating to 7.77 and while the posts with fields or the posts with known lengths still work. This Post with a large size, to allow ongoing posts, has stopped working. In fact I cannot get it to call the readback function at all. 

I have tried changing to Chunked transfer because I went thru the change log and that seems like that might of changed in 7.66,   but I cannot get it to call the read callback.  

What are the conditions to call and more importantly what conditions prevent curl from calling the ReadFunction.

Thanks

**Comment by bagder** [comment, marked as answer]
Moved to https://github.com/curl/curl/issues/7359
